internal/util/clusterextension/clusterextension.go

```diff
@@ -0,0 +1,24 @@
+package clusterextension
+
+import (
+	ocv1 "github.com/operator-framework/operator-controller/api/v1"
+	"github.com/operator-framework/operator-controller/internal/features"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	clusterExtensionWatchNamespaceAnnotation = "olm.operatorframework.io/watch-namespace"
+)
+
+// GetWatchNamespace determines the watch namespace the ClusterExtension should use
+// Note: this is a temporary artifice to enable gated use of single/own namespace install modes
+// for registry+v1 bundles. This will go away once the ClusterExtension API is updated to include
+// (opaque) runtime configuration.
+func GetWatchNamespace(ext *ocv1.ClusterExtension) string {
+	if features.OperatorControllerFeatureGate.Enabled(features.SingleOwnNamespaceInstallSupport) {
+		if ext != nil && ext.Annotations[clusterExtensionWatchNamespaceAnnotation] != "" {
+			return ext.Annotations[clusterExtensionWatchNamespaceAnnotation]
```

Comment:
Should we fail if the annotation value contains a comma so that we don't sign ourselves up for MultiNamespace support?

Reply:
yeah - I was thinking about that. Ultimately, this will go away, though. And all the user will be saying is that it wants the operator to watch namespace "n1,ns2,ns3,ns4" XD. This was just the first iteration to get some tangible experience with the whole thing. I also wanted to run through going from own to single (it worked insofar as the generated role went to the new watch namespace =D)